source/_integrations/smartthings.markdown

```diff
@@ -285,88 +196,24 @@ The SmartThings Scene platform lets you activate scenes defined in SmartThings w
 
 ### Switch
 
-The SmartThings Switch platform lets you control devices that have the [`switch`](https://developer.smartthings.com/docs/devices/capabilities/capabilities-reference#switch) capability that are not already represented by a more specific platform. The following optional capabilities will provide energy and power utilization information:
-
-| Capability                                                                                                      | Switch Features                                         |
-| --------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------- |
-| [`energyMeter`](https://developer.smartthings.com/docs/devices/capabilities/capabilities-reference#energyMeter) | energy consumption (`today_energy_kwh` state attribute) |
-| [`powerMeter`](https://developer.smartthings.com/docs/devices/capabilities/capabilities-reference#powerMeter)   | power consumption (`current_power_w` state attribute)   |
-
-## Troubleshooting
-
-### Setup
-
-#### Aborted: Home Assistant is not configured correctly to receive updates from SmartThings
-
-This error message occurs when you do not have an active Home Assistant Cloud (Nabu Casa) subscription and the external URL is not configured correctly (it must start with `https`). Update your Home Assistant configuration per the prerequisites above, and try again.
+The SmartThings Switch platform lets you control devices that have the [`switch`](https://developer.smartthings.com/docs/devices/capabilities/capabilities-reference#switch) capability that are not already represented by a more specific platform.
 
-#### Error: The token must be in the UID/GUID format
+## Debugging
 
-The personal access token does not match the expected format. Make sure you are copying the entire token and that there are no extraneous characters (such as trailing whitespace) and try again.
+If debug logging is enabled, the integration shows all the received events in the logs.
 
-#### Error: `pysmartthings.errors.APIInvalidGrant: Invalid refresh token:` (after a restore)
+By downloading the device diagnostics (by going to a device page, selecting the 3 dots and downloading diagnostics) it will capture events for that device for 5 seconds, after which it will return a JSON file with the state of the device and the events.
+This can be helpful for diagnosing state updates, for example by selecting the button and then turning on the device physically.
```

Comment:
```suggestion
This can be helpful for diagnosing state updates, for example by selecting the button and then turning on the device physically.

- Follow these steps to [enable debug logs and diagnostics](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics)
```